### PR TITLE
WIP: QuantileLoss

### PR DESCRIPTION
**Pull request by joshday**
Finally starting a PR.  This could be changed to use the same parameterization trick as LPDistLoss.  Thoughts?  

Still needs a UnicodePlot in the docstring.

### CODE REVIEW

**Comment by Evizero** [comment]
> This could be changed to use the same parameterization trick as LPDistLoss.

As a statistician, do you think dispatching on the quantile might be of use? It makes a **lot** of sense for `LPDistLoss`, but not so much for a `EpsilonInsLoss` as far as I can judge, which is why they differ in the way the parameter is handled.

**Comment by joshday** [comment]
Someone may want to dispatch on `QuantileLoss(.5)`, but since this is just proportional to `L1DistLoss`, it's probably not worth it.

File: test/runtests.jl [hunk 1]
```diff
@@ -1,3 +1,4 @@
+module LossFunctionsTests
```
Comment (joshday) [inline test/runtests.jl]: Workflow.  Is that all right?

```julia
reload("LossFunctions")
include("tests/runtests.jl")
```

Comment (joshday) [inline test/runtests.jl]: actually `include("src/LossFunctions.jl"); include("tests/runtests.jl")`

Comment (Evizero) [inline test/runtests.jl]: sure, as long as it doesn't mess with `Pkg.test`.

**Comment by Evizero** [comment]
Can I bully you into adapting those test as well? (or one of those)

- https://github.com/JuliaML/LossFunctions.jl/blob/master/test/tst_properties.jl#L151-L173 (make sure interface works as expected)
- https://github.com/JuliaML/LossFunctions.jl/blob/master/test/tst_loss.jl#L316-L322 (simple naive form reference function)

**Comment by joshday** [comment]
No problem!  Thanks for the pointers.

**Comment by Evizero** [comment]
Thanks! Feel free to merge once the tests pass. I'll adapt the documentation when I add the little preview plot

**Pull request merged**